cmake/FindJavacc.cmake

```diff
@@ -4,6 +4,17 @@ mark_as_advanced(JAVACC_EXECUTABLE)
 if(JAVACC_EXECUTABLE)
   set(JAVACC_FOUND 1)
   message(STATUS "The javacc executable: ${JAVACC_EXECUTABLE}")
+  execute_process(
+      COMMAND "${JAVACC_EXECUTABLE}" --version
+      OUTPUT_VARIABLE JAVACC_TEMP_VERSION
+  )
+  string(REGEX MATCH ".* ([0-9]+\.[0-9]+[\.0-9]*) .*" JAVACC_TEMP_VERSION2_UNUSED "${JAVACC_TEMP_VERSION}")
```

Comment:
This regex matches even to ` 0x0.0 ` :open_mouth:
How about `".* ([0-9]+\\.[0-9]+(\\.[.0-9]+)*) .*"`?
This is similar to CMake's code:  https://github.com/Kitware/CMake/blob/v3.17.2/Tests/CMakeOnly/CheckCXXCompilerFlag/CMakeLists.txt#L36

You and I know that `JAVACC_TEMP_VERSION2_UNUSED` is stupid, so I think `string(REGEX REPLACE ...)` (what you did yesterday and CMake does) is better choice here.
The reason why I proposed `REGEX MATCH` was I wanted to use unreplaced `JAVACC_TEMP_VERSION` again, i.e., `message(STATUS "unexpected output from "javacc --version":\n${JAVACC_TEMP_VERSION}")`. In your code, you don't use unreplaced `JAVACC_TEMP_VERSION`, so there's no reason avoid destructing (replacing) `JAVACC_TEMP_VERSION`.

Reply:
Thanks for the extensive testing and the thinking outside the box (not just the normal version numbers but also some exotic ones).

- the escaping of the special characters (always great fun to get the right syntax).
- Actually I personally think `MATCH` is better as we want to have the match and we don't really replace something. But it is also a matter of taste.
- Thanks for the link to the testing for the Intel compiler version. It shows also that regular expressions can be formulated in a number of ways.